lib/src/command/logout.dart

```diff
@@ -6,20 +6,37 @@ import 'dart:async';
 
 import '../command.dart';
 import '../oauth2.dart' as oauth2;
+import '../tokens.dart';
 
 /// Handles the `logout` pub command.
 class LogoutCommand extends PubCommand {
   @override
   String get name => 'logout';
   @override
-  String get description => 'Log out of pub.dartlang.org.';
+  String get description => 'Log out of pub.dev or any third-party pub server.';
   @override
-  String get invocation => 'pub logout';
+  String get invocation => 'pub logout [<server>] [--all]';
 
-  LogoutCommand();
+  /// Whether to log out of all servers, including third-party pub servers.
+  bool get all => argResults['all'];
```

Comment:
As this is only referenced in the `run()` method, it should not be exposed as public field like this.